Creates Kubecost Cloud Savings library plus Abandoned Workloads for on-prem (#729)
kubecost/docs: kubecost-cloud-cluster-right-sizing.md

# Kubecost Cloud: Cluster Sizing Recommendations

{% hint style="info" %}

> This documentation should only be consulted when using Kubecost Cloud! For information about cluster sizing recommendations for self-hosted Kubecost, see [here](https://docs.kubecost.com/using-kubecost/navigating-the-kubecost-ui/savings/cluster-right-sizing-recommendations).

Reviewer: I know we do this elsewhere, but using relative links (without the base domain) is always preferable because it allows us to change hostname, domain, and anything else without having to do a massive find-and-replace later. In this case, the relative link would be to `/using-kubecost/navigating-the-kubecost-ui/savings/cluster-right-sizing-recommendations`.

Author: It's probably easier to do a direct link to the .md file itself, which is manageable if our content stays in the same repo (which it should).

Author: Due to non-descript file names for some of our articles, locating them is a bit annoying

{% endhint %}

Kubecost Cloud can provide recommendations for right-sizing your clusters to ensure they are configured in the most cost-effective way. Recommendations are available for any and all clusters.

## Viewing cluster recommendations

You can access cluster right-sizing by selecting *Savings* in the left navigation, then select the *Right-size your cluster nodes* panel.

Kubecost will offer two recommendations: simple (uses one node type) and complex (uses two or more node types). Kubecost may hide the complex recommendation when it is more expensive than the simple recommendation, and present a single recommendation instead. These recommendations and their metrics will be displayed in a chart next to your existing configuration in order to compare values like total cost, node count, and usage.

{% hint style="info" %}

> You may see the Total cost of the simple and complex recommendations are both larger than your current cost. This is because when Kubecost Cloud attempts to find the cheapest configuration of nodes to support the existing cluster workload, it does not read the current cost and attempt to minimize it. In this case, Kubecost Cloud is unable to provide an optimized recommendation for your current workload.

Reviewer 2: ```suggestion
You may see the Total cost of the Simple or Complex recommendations being larger than your current cost. This is because when Kubecost Cloud attempts to find the cheapest configuration of nodes to support the existing cluster workload, it does not read the current cost and attempt to minimize it. In this case, Kubecost Cloud is unable to provide an optimized recommendation for your current workload.
```

{% endhint %}

You can toggle on _Show advanced metrics_ to view more details about your cluster resource consumption.

### Configuring your recommendations

Kubecost Cloud provides its right-sizing recommendations based on the characteristics of your cluster. You have the option to edit certain properties to generate relevant recommendations. Select _Filter_ to configure your settings in the Cluster Sizing Settings window.

There are multiple dropdown menus to consider:
* In the Cluster dropdown, you can select the individual cluster you wish to apply right-sizing recommendations to.
* In the Profile dropdown, select the most relevant category of your cluster. You can select _Production_, _Development_, or _High Availability_.
  * Production: Stable cluster activity, will provide some extra space for potential spikes in activity.
  * Development: Cluster can tolerate small amount of instability, will run cluster somewhat close to capacity.
  * High availability: Cluster should avoid instability at all costs, will size cluster with lots of extra space to account for unexpected spikes in activity.

> * In the Architecture dropdown, you should see _x86_, which is currently the only supported architecture type.

Reviewer 2: AFAIK we do support ARM recommendations (and thus `x86`, `ARM` or `Any`), but only for AWS clusters. It is correct that for GCP and Azure clusters we only provide `x86` nodes.

Reviewer 2: The phrasing in the on-prem docs looks correct to me: 
https://docs.kubecost.com/using-kubecost/navigating-the-kubecost-ui/savings/cluster-right-sizing-recommendations#configuring-your-recommendations

```text
In the Architecture dropdown, select either x86 or ARM. You may only see x86 as an option. This is normal. At the moment, ARM architecture recommendations are only supported on AWS clusters.
```

Author: Ah, Ok. Only say x86 when testing, but this makes sense. I'll update
